test/tbb/test_concurrent_queue.cpp

```diff
@@ -199,3 +199,96 @@ TEST_CASE("Test exception in allocation") {
 }
 
 #endif // TBB_USE_EXCEPTIONS
+
+class MinimalisticObject {
+public:
+    struct flag {};
+    static constexpr std::size_t default_obj = 42;
+
+    MinimalisticObject() = delete;
+    MinimalisticObject(flag) : underlying_obj(default_obj) {}
+
+    MinimalisticObject(const MinimalisticObject&) = delete;
+    MinimalisticObject& operator=(const MinimalisticObject&) = delete;
+
+    std::size_t get_obj() const { return underlying_obj; }
+
+protected:
+    std::size_t underlying_obj;
+    friend struct MoveAssignableMinimalisticObject;
+};
+
+constexpr std::size_t MinimalisticObject::default_obj;
+
+struct MoveAssignableMinimalisticObject : MinimalisticObject {
+public:
+    using MinimalisticObject::MinimalisticObject;
+
+    MoveAssignableMinimalisticObject& operator=(MoveAssignableMinimalisticObject&& other) {
+        if (this != &other) {
+            underlying_obj = other.underlying_obj;
+            other.underlying_obj = 0;
+        }
+        return *this;
+    }
+};
+
+template <typename Container>
+void test_basics(Container& container, std::size_t desired_size) {
+    CHECK(!container.empty());
+
+    std::size_t counter = 0;
+    for (auto it = container.unsafe_begin(); it != container.unsafe_end(); ++it) {
+        CHECK(it->get_obj() == MinimalisticObject::default_obj);
+        ++counter;
+    }
+    CHECK(counter == desired_size);
+
+    container.clear();
+    CHECK(container.empty());
+}
+
+template <template <class...> class Container>
+void test_with_minimalistic_objects() {
+    // Test with MinimalisticObject and no pop operations
+    const std::size_t elements_count = 100;
+    {
+        Container<MinimalisticObject> default_container;
+
+        for (std::size_t i = 0; i < elements_count; ++i) {
+            default_container.emplace(MinimalisticObject::flag{});
+        }
+        test_basics(default_container, elements_count);
+    }
+    // Test with MoveAssignableMinimalisticObject with pop operation
+    {
+        Container<MoveAssignableMinimalisticObject> default_container;
+
+        for (std::size_t i = 0; i < elements_count; ++i) {
+            default_container.emplace(MinimalisticObject::flag{});
+        }
+        test_basics(default_container, elements_count);
+
+        // Refill again
+        for (std::size_t i = 0; i < elements_count; ++i) {
+            default_container.emplace(MinimalisticObject::flag{});
+        }
+
+        MoveAssignableMinimalisticObject result(MinimalisticObject::flag{});
+
+        std::size_t element_counter = 0;
+        while(!default_container.empty()) {
+            CHECK(default_container.try_pop(result));
+            ++element_counter;
+        }
+
+        CHECK(element_counter == elements_count);
+        CHECK(default_container.empty());
+    }
+}
+
+//! \brief \ref regression \ref error_guessing
```

Comment:
It looks like requirement test (may be interface?), please consider move test to conformance

Reply:
Moved to conformance test